build/Test.targets

```diff
@@ -16,12 +11,11 @@
       <PathListSeparator Condition=" '$(OSName)' == 'win' ">%3b</PathListSeparator>
       <ExecPath>$(Stage2Directory)$(PathListSeparator)$(PATH)</ExecPath>
       <ExecPath Condition=" '$(OSName)' == 'win' ">$(Stage2Directory)</ExecPath>
-      <TestArtifactsEnv>$(TestArtifactsDir)</TestArtifactsEnv>
     </PropertyGroup>
```

Comment:
At a minimum, move the property group to be a direct descendant of <Project>. No need for them to be evaluated in a target, as far as I can see.